### PR TITLE
refactor: update to work with resonite

### PR DESCRIPTION
**Pull request by stiefeljackal**
* feat: change from Neos to Resonite

* docs: update to Resonite terminology

* chore: add editor config for space consistency

### CODE REVIEW

File: GenericTypeType.Tests/GenericTypeType.Tests.csproj [hunk 1]
```diff
@@ -9,28 +9,31 @@
   </PropertyGroup>
 
   <PropertyGroup>
-    <NeosDirectory Condition="'$(NeosDirectory)' == ''">C:\Program Files (x86)\Steam\steamapps\common\NeosVR</NeosDirectory>
-    <NeosCoreLibs>$(NeosDirectory)\Neos_Data\Managed</NeosCoreLibs>
-    <NeosNmlLibs>$(NeosDirectory)\nml_libs</NeosNmlLibs>
-    <NeosLibs>$(NeosDirectory)\Libraries</NeosLibs>
+	<ResonitePath>$(MSBuildThisFileDirectory)NeosVR</ResonitePath>
```
Comment (Earthmark) [inline GenericTypeType.Tests/GenericTypeType.Tests.csproj:12]: you make my indentation sad.

File: GenericTypeType.Tests/WorkerManagerGetTypePatchTests.cs [hunk 2]
```diff
@@ -20,7 +20,9 @@ public HarmonyRewriteSetup()
   }
 }
 
-public class WorkerManagerGetTypePatchTests : IClassFixture<HarmonyRewriteSetup>
+#pragma warning disable xUnit1033 // Test classes decorated with 'Xunit.IClassFixture<TFixture>' or 'Xunit.ICollectionFixture<TFixture>' should add a constructor argument of type TFixture
+public class WorkerManagerGetTypePatchTests : IClassFixture<HarmonyRewriteSetup>
```
Comment (Earthmark) [inline GenericTypeType.Tests/WorkerManagerGetTypePatchTests.cs:24]: Does this test still work?

**Comment by Earthmark** [comment]
I spoke with them in Discord and will take over the branch to do the local changes.

File: GenericTypeType.Tests/WorkerManagerGetTypePatchTests.cs (hunk 2 above)
Comment (stiefeljackal) [reply GenericTypeType.Tests/WorkerManagerGetTypePatchTests.cs:24]: Test should still work. I ran the unit tests and made sure they succeeded before making the PR.

File: GenericTypeType.Tests/GenericTypeType.Tests.csproj (hunk 1 above)
Comment (stiefeljackal) [reply GenericTypeType.Tests/GenericTypeType.Tests.csproj:12]: Oh geez. I'm sorry about that. It should've been

```
<PropertyGroup>
  <ResonitePath>...
```

I thought I made sure the indentation was good, but I missed this one for some reason.

**Comment by Earthmark** [comment]
Nevermind on changes, I was concerned about the unit test changes but the warnings are implemented just below, not sure why the suppression was needed. Looks good.

**Pull request merged**